src/js/products.js

```diff
@@ -0,0 +1,129 @@
+
+Handlebars.registerHelper('ifFirstIndex', function (index, options) {
+    if (index == 0) {
+        return options.fn(this);
+    } else {
+        return options.inverse(this);
+    }
+});
+// --------------------------------------------- //
+// fetching the category on product listing
+// --------------------------------------------- //
+var prod_categoriesXHttp = new XMLHttpRequest();
+prod_categoriesXHttp.open("GET", "categories", true);
+prod_categoriesXHttp.onload = function () {
+    if (prod_categoriesXHttp.status >= 200 && prod_categoriesXHttp.status < 400) {
+        var data = JSON.parse(prod_categoriesXHttp.responseText);
+        createProdCategoryContainer(data);
+    } else {
+        console.log("We conected to the server, but it returned an error.");
+    }
+}
+prod_categoriesXHttp.onerror = function () {
+    console.log("Connection Error");
+}
+prod_categoriesXHttp.send();
+
+function createProdCategoryContainer(data) {
+    var slideshowTemplates = document.getElementById("productMenuTemplates").innerHTML;
+    var slideshowCompiledTemplate = Handlebars.compile(slideshowTemplates);
```

Comment:
Compiled templates should be cached for the entire run of page. Ideally compilation should happen during the build process.

Reply:
DONE